contrib/scripts/run-gh-ginkgo-workflow.sh

```diff
@@ -0,0 +1,335 @@
+
+    case "${kernel_tag}" in
+        bpf-next-*)
+            K8S_NODES=3
+            NETNEXT=1
+            KERNEL=net-next
+            KUBEPROXY=0
+            NO_CILIUM_ON_NODES=kind-worker2
+            ;;
+        4.19-*)
+            KERNEL=419
+            ;;
+        5.4-*)
+            KERNEL=54
+            ;;
+        *)
+            usage 1
+            ;;
+    esac
+
+    if [[ -z "${commit_sha}" ]]; then
+        commit_sha="$(git rev-parse HEAD)"
+    fi
+    quay_org="cilium"
+
+    # Let's run the tests.
+    #
+    # GitHub actions do not support IPv6 connectivity to outside world. If the
+    # infrastructure environment supports it, then CILIUM_NO_IPV6_OUTSIDE can
+    # be removed
+    ${connect} -t "cd /host/test; \
+        K8S_VERSION=${K8S_VERSION} \
+        K8S_NODES=${K8S_NODES} \
+        NETNEXT=${NETNEXT} \
+        KERNEL=${KERNEL} \
+        KUBEPROXY=${KUBEPROXY} \
+        NO_CILIUM_ON_NODES=${NO_CILIUM_ON_NODES} \
+        CNI_INTEGRATION=kind \
+        INTEGRATION_TESTS=true \
+        CILIUM_NO_IPV6_OUTSIDE=true \
+        ./test.test \
+            --ginkgo.focus=\"${cli_focus}\" \
+            --ginkgo.skip= \
+            --ginkgo.seed=1679952881 \
+            --ginkgo.v -- \
+            -cilium.provision=false \
+            -cilium.image=quay.io/${quay_org}/cilium-ci \
+            -cilium.tag=${commit_sha}  \
+            -cilium.operator-image=quay.io/${quay_org}/operator \
+            -cilium.operator-tag=${commit_sha} \
+            -cilium.hubble-relay-image=quay.io/${quay_org}/hubble-relay-ci \
+            -cilium.hubble-relay-tag=${commit_sha} \
+            -cilium.kubeconfig=/root/.kube/config \
+            -cilium.provision-k8s=false \
+            -cilium.operator-suffix=-ci \
+            -cilium.holdEnvironment=true"
+}
+
+# Parse options
+
+commit_sha=""
+cli_focus="K8s"
+kernel_version=""
+k8s_version=""
+skip_to_tests=""
+tmpdir="/tmp"
+wakeup_vm=""
+OPTIND=1
+while getopts "c:f:hl:k:st:w" opt; do
+    case "$opt" in
+        c)
+            commit_sha="${OPTARG}"
+            ;;
+        f)
+            cli_focus="${OPTARG}"
+            ;;
+        h)
+            usage 0
+            ;;
+        l)
+            kernel_version="${OPTARG}"
+            ;;
+        k)
+            k8s_version="${OPTARG}"
+            ;;
+        s)
+            skip_to_tests=true
+            ;;
+        t)
+            tmpdir="${OPTARG}"
+            ;;
+        w)
+            wakeup_vm=true
+            ;;
+        *)
+            usage 1
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+# Local dependencies
+
+check_cmd_or_fail git zstd
+root_dir="$(git rev-parse --show-toplevel)"
+
+helm="helm"
+if ! check_cmd "${helm}"; then
+    install_helm
+    helm="${tmpdir}/helm"
+fi
+
+if ! check_cmd ginkgo; then
+    install_ginkgo
+fi
+
+yq="yq"
+if ! check_cmd "${yq}"; then
+    yq="docker run --rm -v ${PWD}:/workdir --user $(id -u):$(id -g) mikefarah/yq:4.27.3"
```

Comment:
do we need a renovate reference to auto-update this image?

Reply:
Probably not this one, as long as the few basic `yq` invocations we do in the script work fine, it probably doesn't matter what version we're using.

Reply:
:grin: I ran into this while testing - probably best not to use local `yq` at all, since there's competing incompatible implementations apparently: 
- https://github.com/kislyuk/yq
- https://github.com/mikefarah/yq

Reply:
Oh no :facepalm:
OK I'll update to just use the Docker container.